### PR TITLE
cpu/fe310: RTC: use rtc_mktime()

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Use RTC helper functions instead of libc functions.
This gives us y2038 safety by the extended epoch and saves a good chunk of memory:

#### picolibc mktime():

```
   text	   data	    bss	    dec	    hex	filename
  15048	    520	   2504	  18072	   4698	tests/periph_rtc/bin/hifive1/tests_periph_rtc.elf
```

#### rtc_mktime():

```
   text	   data	    bss	    dec	    hex	filename
   7632	     40	   2452	  10124	   278c	tests/periph_rtc/bin/hifive1/tests_periph_rtc.elf
```

### Testing procedure

`tests/periph_rtc` should still work.
(untested, I don't have the hardware)

### Issues/PRs references

#13277

### CODE REVIEW

File: sys/picolibc_syscalls_default/syscalls.c [hunk 1]
```diff
@@ -24,6 +24,7 @@
 #include <stdio.h>
 #include <sys/times.h>
 
+#include "irq_arch.h"
```
Comment (aabadie) [inline sys/picolibc_syscalls_default/syscalls.c:27]: This seems unrelated. Can you explain ?

Comment (benpicco) [reply sys/picolibc_syscalls_default/syscalls.c:27]: Oh I already moved that to a different PR #14901

But this PR will be obsolete with #14895 anyway.

Comment (aabadie) [reply sys/picolibc_syscalls_default/syscalls.c:27]: Hmm ok, and #14895 should be merged before Y2038 ;)

**Comment by aabadie** [comment]
Just tested on hifive1b and the test application is failing:

```
RIOT RTC low-level driver test
This test will display 'Alarm!' every 2 seconds for 4 times
  Setting clock to   2020-02-28 23:59:57
Clock value is now   2020-02-29 00:00:05
```

The time is not set correctly.

**Comment by aabadie** [comment]
It's also completely broken on master...

**Comment by aabadie** [comment]
I found the culprit: 442b11d0eedef5657cc3646b5765eda55e205ae7 from #14741

**Review by aabadie — APPROVED** [review]
With this patch:

```diff
diff --git a/cpu/fe310/include/periph_cpu.h b/cpu/fe310/include/periph_cpu.h
index dcc468fce1..18d7af039f 100644
--- a/cpu/fe310/include/periph_cpu.h
+++ b/cpu/fe310/include/periph_cpu.h
@@ -173,7 +173,7 @@ typedef struct {
 #define RTT_MIN_FREQUENCY   (1U)                    /* in Hz */
 
 #ifndef RTT_FREQUENCY
-#define RTT_FREQUENCY       (RTT_MAX_FREQUENCY)     /* in Hz */
+#define RTT_FREQUENCY       (RTT_MIN_FREQUENCY)     /* in Hz */
#
```
I'm able to have periph_rtc to work (it reverts the situation to what it was before #14741 was merged).
So it's not worse than master (it fails badly) but it also works the same as master under the same conditions, when master was still working.

I also confirm the code size decrease compared to master (from 21419B to 12102B for `tests/periph_rtc`).

ACK
